File: Blind_75_LeetCode_Questions/Product_of_Array_Except_Self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums)
    {
        int n = nums.size();
        vector<int> result(n);
        vector<int> temp_arr(n);
        vector<int> temp(n);
        for(int i = 0; i < nums.size(); i++)
        {
            temp_arr[i] = 1;
            if(temp_arr[i] == nums[i])
            {
               temp[i] = nums[i+1] * nums[i+2]; 
            }
            else 
            {
                temp[i] = nums[i] * nums[i+1];
                result[i] = temp_arr[i+1] * nums[i];
    
            }
        }
        return result;
    }
};
// ERROR
// Tried to make switch case with if statement, but getting worse.
// Fix tomorrow.

File: Blind_75_LeetCode_Questions/java/Array/Q238_Product_of_Array_Except_Self/Product_of_Array_Except_Self.java
class Solution {
    public int[] productExceptSelf(int[] nums) 
    {
        int len = nums.length;
        int[] left = new int[len];
        int[] right = new int[len];
        int[] answer = new int[len];

        left[0] = 1;
        right[len - 1] = 1;

        for(int i = 1; i < len; ++i)
        {
            left[i] = left[i - 1] * nums[i - 1];
        }

        for(int i = len - 2; i >= 0; --i)
        {
            right[i] = right[i + 1] * nums[i + 1];
        }

        for(int i = 0; i < len; ++i)
        {
            answer[i] = left[i] * right[i];
        }
        return answer;
        
    }
}
